src/universe/index.ts

```diff
@@ -460,24 +466,30 @@ export class Universe extends APICarrier {
   private mqtt: realtime.RealtimeClient | null = null
   public base: string
   public universeBase: string
+  public universeHost: string = 'hello-charles.com'
   public mqttUniverseBase: string
   private static readonly endpoint: string = 'api/v0/universes'
   private _cachedMeData?: MeData
 
   public constructor (options: UniverseOptions) {
+    // NOTE: there is a bit unfortunate duplication necessary because of https://github.com/microsoft/TypeScript/issues/8277
+    // not being able to declare const before super calls
     super({
       injectables: {
         // symmetric to the legacy implementation below
-        base: options.universeBase ?? `https://${options.name}.hello-charles.com`
+        base: options.universeBase ?? `https://${options.name}.${options.universeHost ?? 'hello-charles.com'}`
```

Comment:
I'd suggest to move this `hello-charles.com` to some `const DEFAULT_UNIVERSE_HOST`

Reply:
does not work because of the TS compiler unfortunately, see ref in code or https://github.com/microsoft/TypeScript/issues/8277